src/types/index.d.ts

```diff
@@ -238,33 +215,59 @@ export interface EnvoyVersion {
   build: string
   kumaDpCompatible?: boolean
 }
+export interface Version {
+  kumaDp: KumaDpVersion
+  kumaCp?: KumaCpVersion
+
+  envoy: EnvoyVersion
+
+  dependencies: Record<string, string>
+}
+
+export interface DiscoveryServiceStats {
+  responsesSent?: number
+  responsesAcknowledged?: number
+  responsesRejected?: number
+}
 
-export interface DiscoverySubscriptionStatus {
+export type SubscriptionStatus = {
   lastUpdateTime: string
   total: DiscoveryServiceStats
+}
+
+export type KDSSubscriptionStatus = {
+  stat: Record<string, DiscoveryServiceStats>
+} & SubscriptionStatus
+
+export type DiscoverySubscriptionStatus = {
   cds: DiscoveryServiceStats
   eds: DiscoveryServiceStats
   lds: DiscoveryServiceStats
   rds: DiscoveryServiceStats
+} & SubscriptionStatus
+
+export type Subscription = {
+  id: string
+  connectTime?: string
+  disconnectTime?: string
 }
 
-export interface Version {
-  kumaDp: KumaDpVersion
+export type KDSSubscription = {
+  version: any
 
-  envoy: EnvoyVersion
+  config: string
+  zoneInstanceId?: string
+  globalInstanceId?: string
+  status: KDSSubscriptionStatus
+} & Subscription
 
-  dependencies: Record<string, string>
-}
+export type DiscoverySubscription = {
+  version?: Version
 
-export interface DiscoverySubscription {
-  id: string
   controlPlaneInstanceId: string
-  connectTime?: string
-  disconnectTime?: string
-  status: DiscoverySubscriptionStatus
   generation?: number
-  version?: Version
-}
+  status: DiscoverySubscriptionStatus
+} & Subscription
```

Comment:
These are the old handwritten types that we still need to support until we get these types OpenAPI'ed.

I found these to be slightly wrong, and I also moved them together in the file so I could look at the in the same place without a split. Whenever you are working on KDSSubscriptions you invariably need to work on DiscoverySubscription , much better if they are next to each other.